test/suites/upgrade_ostree.robot

```diff
@@ -0,0 +1,99 @@
+*** Settings ***
+Documentation       Tests related to MicroShift upgradeability on OSTree-based systems
+
+Resource            ../resources/common.resource
+Resource            ../resources/oc.resource
+Resource            ../resources/ostree.resource
+Library             Collections
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Force Tags          ostree
+
+
+*** Variables ***
+${USHIFT_HOST}      ${EMPTY}
+${USHIFT_USER}      ${EMPTY}
+
+
+*** Test Cases ***
+Rebooting Healthy System Should Result In Data Backup
+    [Documentation]    Check if rebooting healthy system will result in backing up of MicroShift data
+
+    Wait Until Greenboot Health Checks Are Done
+    System Should Be Healthy
+    Remove Existing Backup For Current Deployment
+
+    Reboot
+    Wait Until MicroShift Is Ready
+
+    Status Of Current Boot Pre Run Should Be Success
+    Backup For Booted Deployment Should Exist
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Test suite setup
+    Check Required Env Variables
+    Login MicroShift Host
+
+Teardown
+    [Documentation]    Test suite teardown
+    Logout MicroShift Host
+
+Wait Until Greenboot Health Checks Are Done
+    [Documentation]    Wait until greenboot healthchecks are done
+
+    Wait Until Keyword Succeeds    5m    15s
+    ...    Check Service Property    greenboot-healthcheck.service    SubState    exited
+
+Wait Until MicroShift Is Ready
+    [Documentation]    Wait until MicroShift service is ready
+
+    Wait Until Keyword Succeeds    2m    15s
+    ...    Check Service Property    microshift.service    ActiveState    active
+
+Check Service Property
+    [Documentation]    Checks if property of a service has expected value
+    [Arguments]    ${service}    ${property}    ${expected_value}
+
+    ${value}    ${rc}=    Execute Command
+    ...    systemctl show ${service} -P ${property}
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+    Should Be Equal As Strings    ${expected_value}    ${value}
+
+Reboot
```

Comment:
From @dhellmann ([src](https://github.com/openshift/microshift/pull/1826#discussion_r1212161152))
> This could go into the microshift-host resource file.